Layer/Pdo/Pdo.php

```diff
@@ -194,7 +194,12 @@ public function lastInsertId($name = null)
     public function prepare($statement, array $options = [])
     {
         if (!isset($options[\PDO::ATTR_CURSOR])) {
-            $options[\PDO::ATTR_CURSOR] = \PDO::CURSOR_SCROLL;
+            try {
+                $this->getConnection()->getAttribute(\PDO::ATTR_CURSOR);
+                $options[\PDO::ATTR_CURSOR] = \PDO::CURSOR_SCROLL;
```

Comment:
> Also, if this patch works, then this is a bug in PDO with SQLite driver and we should open a bug on bugs.php.net, do you agree?

Yes.

> Hmm, I will try. However, iterator support will no be longer be available if scroll cursor is not enabled. Should we throw an error then?

Forward only iterator will be available. Scrollable cursor is used to customize the fetch orientation.

Throw an error if the user want to use a feature which is not supported by the driver is interesting but we have to re-think the way we prepare our statement.
Currently we enable the scroll cursor before the user has chosen if he want to use the feature so we can't throw an error according to his choice.